docs/testing/testing-overview.md

```diff
@@ -0,0 +1,138 @@
+* __Component tests__ (~50%, business analysts & QA) – part of CI.
+* __Integration tests__ (~20%, architects) – they should care only about a couple of layers at the time, executed once a night in dev environment.
+* __System tests__ (~10%, Business analysts & architects) – run in staging, at that level you can test 3rd parties.
+* __Manual tests__ (~5%, people) – exploratory, you never follow a handbook.
+
+All of this might feel confusing and overwhelming even for seasoned developers. That's why the short talk [What We Can Learn About Testing From The Wheel](https://www.youtube.com/watch?v=Da9wfQ0frGA) by _Kent C. Dodds_ from _Ignite Fluent 2016_ might help understand differences. The author uses a car metaphor to make it easier to distinguish some of the most popular types of testing.
+
+Next paragraphs explain how all those different types of testing are used to keep Calypso healthy.
+
+### Server side tests
+
+This test configuration contains unit tests that verify code located in `server` top level folder. 
+
+It supports automatic test discovery. We only need to put a test file into a `test` subfolder, next to the files we want to test.
+
+Tests can be run in 3 different modes:
+```bash
+> # run the entire server suite
+> npm run test-server
+> # run a configuration customized to work with continuous integration 
+> npm run test-server:ci
+> # run tests in watch mode, by default it executes tests for the modified files only
+> npm run test-client:watch
+```
+
+Those tests are executed on every push on continuous integration (we use CircleCi). This is why all individual tests need to be blazing fast. Please note that network connection is disabled for this configuration.
+
+_Check also how to write [unit components](unit-tests.md)._
+
+### Client side tests
+
+This test configuration contains unit and component tests that verify code located in `client` top level folder. 
+
+It supports automatic test discovery. We only need to put a test file into a `test` subfolder, next to the files we want to test. 
+
+Tests can be run in 3 different modes:
+```bash
+> # run the entire client suite
+> npm run test-client
+> # run a configuration customized to work with continuous integration
+> npm run test-server:ci
+> # run tests in watch mode, by default it executes tests for the modified files only
+> npm run test-client:watch
+```
+
+They are executed on every push on continuous integration (CircleCI). This is why all individual tests need to be blazing fast. Please note that network connection is disabled for this configuration.
+
+_Check also how to write [unit tests](unit-tests.md) and [component tests](component-tests.md)._
+
+### Integration tests
+
+This test configuration contains integration tests that verify code located in `bin`, `client`, `server` and `test` top level folders. They should test how a group of components or a larger part of business logic works together.
```

Comment:
I think we might want to note that this is meant for services/flows that need to be up.

Anything visual is better done as a full e2e test in https://github.com/Automattic/wp-e2e-tests